bundles/org.openhab.binding.resol/src/main/java/org/openhab/binding/resol/internal/discovery/ResolVBusBridgeDiscovery.java

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.resol.internal.discovery;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.resol.internal.ResolBindingConstants;
+import org.openhab.core.config.discovery.AbstractDiscoveryService;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.config.discovery.DiscoveryService;
+import org.openhab.core.thing.ThingUID;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import de.resol.vbus.TcpDataSource;
+import de.resol.vbus.TcpDataSourceProvider;
+
+/**
+ * The {@link ResolVBusBridgeDiscovery} class provides the DiscoverySerivce to
+ * discover Resol VBus-LAN adapters
+ *
+ * @author Raphael Mack - Initial contribution
+ */
+@Component(service = DiscoveryService.class)
+@NonNullByDefault
+public class ResolVBusBridgeDiscovery extends AbstractDiscoveryService {
+    public static final String THING_PROPERTY_IPADDRESS = "ipAddress";
+    public static final String THING_PROPERTY_PORT = "port";
+    public static final String THING_PROPERTY_ADAPTER_SERIAL = "adapterSerial";
+
+    private final Logger logger = LoggerFactory.getLogger(ResolVBusBridgeDiscovery.class);
+
+    private volatile boolean discoveryRunning = false;
+
+    public ResolVBusBridgeDiscovery() throws IllegalArgumentException {
+        super(ResolBindingConstants.SUPPORTED_BRIDGE_THING_TYPES_UIDS, 35, false);
+    }
+
+    @Override
+    protected void startScan() {
+        discoveryRunning = true;
+        scheduler.execute(this::searchRunnable);
```

Comment:
Can you point me to an example? Since I was requested to use directly the execute() with the method as parameter I don't have any object anymore that could be canceled...

Reply:
If you use `submit` instead then it would return a Future which you could store to a field and then cancel that future as part of `stopScan`.